libraries/AC_Fence/AC_Fence.h

```diff
@@ -194,6 +215,7 @@ class AC_Fence
     AP_Int8         _total;                 // number of polygon points saved in eeprom
     AP_Int8         _ret_rally;             // return to fence return point or rally point/home
     AP_Int16        _ret_altitude;          // return to this altitude
+    AP_Enum<Location::AltFrame> _alt_type;  // altitude type
```

Comment:
```suggestion
    AP_Enum<Location::AltFrame> _alt_frame;  // altitude frame
```

We really should be using "frame" in the code.

I contend we should really be using FRAME as the parameter, too.

Reply:
@peterbarker, I disagree with simply renaming this parameter to "_FRAME" because:

- We're already using "_TYPE" in other parameter names (e.g. RTL_ALT_TYPE and FOLL_ALT_TYPE) and in my previous review I asked that _FRAME be changed to _TYPE.
- It will be confusing for developers because the enumerations of the variable will not match the code.

If we make the enumerations the same as in the code and agree (at some point) to rename the other two existing variables then I'm happy with this.

I think for now we should just let the name remain as _ALT_TYPE.

Reply:
> @peterbarker, I disagree with simply renaming this parameter to "_FRAME" because:
> 
>     * We're already using "_TYPE" in other parameter names (e.g. RTL_ALT_TYPE and FOLL_ALT_TYPE) and in my previous review I asked that _FRAME be changed to _TYPE.

Apologies to to Stephanie for requesting the opposite - comes from not reading through the history, I guess.

>     * It will be confusing for developers because the enumerations of the variable will not match the code.

It's already confusing as the type of the containing variable is explicitly Frame, and all the methods on location say "frame" not "type"

> If we make the enumerations the same as in the code and agree (at some point) to rename the other two existing variables then I'm happy with this.
> 
> I think for now we should just let the name remain as _ALT_TYPE.

I'm quite happy to create a follow-up PR that changes all of them to `FRAME` so we don't hold this one up.  That'll be unnecessary churn, but I kind of feel bad for requesting a change back to FRAME on this PR.

Reply:
As part of that change we would need to do the param conversion (at least for RTL_ALT_TYPE) and we'd need to test it of course.  Not trying to be annoying but the testing is often what delays things longer than we'd perhaps at first expect.